logstash-core/lib/logstash/runner.rb

```diff
@@ -149,8 +149,12 @@ def execute
       return 1
     end
 
-    # Print a warning to STDERR for bad java versions
-    LogStash::Util::JavaVersion.warn_on_bad_java_version
+    # Exit on bad java versions
+    java_version = LogStash::Util::JavaVersion.version
+    if LogStash::Util::JavaVersion.bad_java_version?(java_version)
+      $stderr.puts "Java version 1.8.0 or later is required. (You are running: #{java_version})"
+      return 1
+    end  
```

Comment:
Shouldn't we use `signal_usage_error` and `I18n` for the string?

Like this https://github.com/elastic/logstash/pull/5243/files#diff-da023432533015a65b7931c1c17d4c60R177

`signal_usage_error` will exit with the right status code.